### PR TITLE
issue289 : Investigate PowerMock code coverage issue 

### PR DESCRIPTION
**Pull request by ndegwamartin**
I upgraded a few libraries related to the testing environment however the real fix was to exclude any classes under test from being included in the PrepareForTest annotation. This is because from Powermock's perspective if you add a Class to the PrepareForTest annotation you are asking it to mock that class , and mocked classes (according to Jacoco/PowerMock setup) do not require code coverage so they are excluded by the framework.

Minor refactoring of the org.opensrp.scheduler.service.ActionServiceTest class demonstrates the workaround fix;

### CODE REVIEW

File: opensrp-core/pom.xml [hunk 1]
```diff
@@ -1,228 +1,219 @@
...
+	<build>
+		<resources>
+			<resource>
+				<directory>../assets/config</directory>
+			</resource>
+			<resource>
+				<directory>../assets/schedules</directory>
+				<targetPath>schedules</targetPath>
+			</resource>
+			<!-- <resource> <directory>../assets/form</directory> <targetPath>form</targetPath>
```
Comment (pld) [inline opensrp-core/pom.xml]: why not remove this?

File: opensrp-core/pom.xml [hunk 2]
```diff
@@ -1,228 +1,219 @@
...
+			<artifactId>motech-platform-server-api</artifactId>
+			<groupId>org.motechproject</groupId>
+			<version>${motech.version}</version>
+		</dependency>
+		<dependency>
+			<groupId>com.github.ldriscoll</groupId>
+			<artifactId>ektorplucene</artifactId>
+			<version>0.2.0</version>
+		</dependency>
+		<!-- <dependency> <groupId>com.fasterxml.jackson.core</groupId> <artifactId>jackson-core</artifactId>
```
Comment (pld) [inline opensrp-core/pom.xml]: why not remove this?

File: opensrp-core/pom.xml [hunk 3]
```diff
@@ -1,228 +1,219 @@
...
-            <artifactId>spring-data-jpa</artifactId>
-            <version>1.1.0.RELEASE</version>
-        </dependency>
+		<dependency>
+			<groupId>com.fasterxml.jackson.core</groupId>
+			<artifactId>jackson-databind</artifactId>
+			<version>2.2.2</version>
+		</dependency>
 
+		<!-- <dependency> <groupId>com.fasterxml.jackson.core</groupId> <artifactId>jackson-annotations</artifactId>
```
Comment (pld) [inline opensrp-core/pom.xml]: why not remove this?

**Comment by ndegwamartin** [comment]
@sohelsarder I've added the gotcha here https://smartregister.atlassian.net/wiki/spaces/Documentation/pages/65570409/Gotchas

**Pull request merged**